lib/collector_syncronization.rb

```diff
@@ -0,0 +1,113 @@
+require 'uri'
+require 'securerandom'
+
+require 'global_configuration'
+require 'logging'
+require 'vsphere_session'
+require 'metrics_collector'
+require 'infrastructure_collector'
+require 'inventory_collector'
+require 'uc6_connector'
+require 'interval_time'
+
+class CollectorSyncronization
+  using IntervalTime
+  include GlobalConfiguration
+  include Logging
+  include VSphere
+
+  def initialize
+    @environment = ENV['METER_ENV'] || 'development'
+    @configuration = GlobalConfiguration::GlobalConfig.instance
+  end
+
+  def sync_data
+    start_sync if access_granted?
+  end
+
+  private
+
+  def access_granted?
+    @configuration[:verified_api_connection] && @configuration[:verified_vsphere_connection]
+  end
+
+  def start_sync
+    begin
+      logger.debug "On Start Sync"
+      @uc6_connector = UC6Connector.new
+      logger.debug "UC6 connector #{@uc6_connector.inspect}"
+      collect_infrastructures
+      submit_infrastructures
+      collect_machine_inventory
+      sync_remote_ids
+    rescue StandardError => e
+      logger.error e
+      logger.error e.backtrace.join("\n")
+      if ( e.is_a?(RestClient::Exception) )
+        logger.error e.to_s
+        logger.error e.http_body
+      end
+    end
+  end
+
+  def collect_infrastructures
+    logger.debug "On collect insfrastructures"
+    infrastructures = Infrastructure.all
+    InfrastructureCollector.new.run
+    if Infrastructure.empty?
+      logger.debug "No infrastructures discovered"
+    else
+      logger.debug "#{infrastructures.count} infrastructure#{'s' if infrastructures.count > 1} discovered"
+    end
+  end
+
+  def submit_infrastructures
+    logger.info "On submit infrastructures"
```

Comment:
I would use single quotes here since there is no variable interpolation within the string